pkg/dns/ibm/private/dnssvcs_provider.go

```diff
@@ -0,0 +1,273 @@
+package private
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	dnssvcsv1 "github.com/IBM/networking-go-sdk/dnssvcsv1"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	dnsclient "github.com/openshift/cluster-ingress-operator/pkg/dns/ibm/private/client"
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	iov1 "github.com/openshift/api/operatoringress/v1"
+)
+
+var (
+	_                       dns.Provider = &Provider{}
+	log                                  = logf.Logger.WithName("dns")
+	defaultDNSSVCSRecordTTL              = int64(120)
+)
+
+type Provider struct {
+	dnsService dnsclient.DnsClient
```

Comment:
In case of CIS sdk, zoneID is stored and used in the provider options which is set during client creation. So we decided to create a zoneID to client map, so we don't have to create a client each time there is a dns operation. https://github.com/SzucsAti/cluster-ingress-operator/blob/ibm-dnssvc-support/pkg/dns/ibm/public/cis_provider.go#L45-#L52

In case of DNS-SVCS sdk, zoneID is a required parameter in every function and it's not needed to set in client options. https://github.com/SzucsAti/cluster-ingress-operator/blob/ibm-dnssvc-support/pkg/dns/ibm/private/dnssvcs_provider.go#L44-#L49

Reply:
Got it, thanks for the explanation.